staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -297,6 +298,12 @@ func NewReflectorWithOptions(lw ListerWatcher, expectedType interface{}, store R
 	}
 
 	r.useWatchList = clientfeatures.FeatureGates().Enabled(clientfeatures.WatchListClient)
+	if r.useWatchList {
```

Comment:
I'm aligned with that (especially to avoid redoing all fake-client tests to work with watch-list).